### DISCUSSION TITLE
How to include hidden directory in a search

### DISCUSSION THREAD

**Discussion by sarmong** [BurntSushi/ripgrep#2368] [Q&A]
I want to run `rg` with `--no-hidden`, but want to include `.config` and `.local` in a search. I was trying to do this with `--glob` to no avail.

**Comment by BurntSushi** [comment]
Here's some setup:

```
$ mkdir .config .local .other                               
$ echo foo > {.config/test,.local/test,.other/test}
```

Here's one way to search `.config` and `.local` with the `-g/--glob` flag:

```
$ rg -g '{.config,.config/**,.local,.local/**}' foo 
.config/test
1:foo

.local/test
1:foo
```

This is indeed a bit weird. The issue is that `.config` matches `.config`, but nothing else, including `.config/test`. And `.config/**` matches `.config/test` but not `.config`. So you need both rules.

However, you can create a `.rgignore` file and whitelist these two directories. At which point, ripgrep will automatically search in them:

```
$ echo '!.config\n!.local\n' > .rgignore
$ rg foo
.config/test
1:foo

.local/test
1:foo
```

Why doesn't `.rgignore` also need the `.config/**` rule? The reason is that when you use `-g/--glob`, at least one of the globs _must_ match the file path in order to search it. Otherwise, something like `-g '*.py'` would not behave as one would expect.
